### DISCUSSION TITLE
How to execute script before Next 13 start rendering?

### DISCUSSION THREAD

**Discussion by nucleartux** [vercel/next.js#48594] [Help]
### Summary

I have third party script which sets some information in the `window` object.
I need this variables to set before start of execution of any code.
How I can achieve that?
I tried to use this code:
```
 <Script src="https://third.party/script.js" strategy="beforeInteractive" />
```
But it actually runs after NextJS code.

**Comment by marlonschlosshauer** [comment]
Using the next `<Script>` component with `strategy="beforeInteractive"` didn't work for me on Next 13. Instead I had to go back to `<script>`, with `defer` and `async` (and putting it inside of the `head` instead  of `body`/`html`). 

So my layout render return looks like this: 

```html
  <html>
      <head>
          <script defer async src="https://third.party/script.js" />
      </head>
      <body>{children}</body>
  </html>
```
